framework/base/Module.php

```diff
@@ -571,32 +571,43 @@ public function createController($route)
             return false;
         }
 
-        if (strpos($route, '/') !== false) {
-            list($id, $route) = explode('/', $route, 2);
-        } else {
-            $id = $route;
-            $route = '';
+        // module and controller map take precedence
+        $routeInfo=pathinfo($route);
```

Comment:
Route here is `moduleId/controllerId/actionId`.

- moduleId and actionId are `[a-z-]+`.
- controllerId is `([a-z0-9_/]+/)?[a-z-]+`.

So for `moduleId/prefix/controllerId/actionId` it gives 

```
 'dirname' =>
  string(28) "moduleId/prefix/controllerId"
  'basename' =>
  string(8) "actionId"
  'filename' =>
  string(8) "actionId"
```

Reply:
Yes, absolutely.